File: tests/prebuilt_classes.py
from models import WoWPlayer


### Tanks
def TankPaladin(name: str, offhealer=False, offdps=False) -> WoWPlayer:
    return WoWPlayer(
        name=name,
        tankMain=True,
        hasBrez=True,
        hasLust=False,
        offhealer=offhealer,
        offdps=offdps,
    )


def TankWarrior(name: str, offdps=False) -> WoWPlayer:
    return WoWPlayer(
        name=name, tankMain=True, hasBrez=False, hasLust=False, offdps=offdps
    )


def TankDruid(name: str, offhealer=False, offdps=False) -> WoWPlayer:
    return WoWPlayer(
        name=name,
        tankMain=True,
        hasBrez=True,
        hasLust=False,
        offhealer=offhealer,
        offdps=offdps,
    )


def TankDeathKnight(name: str, offdps=False) -> WoWPlayer:
    return WoWPlayer(
        name=name, tankMain=True, hasBrez=True, hasLust=False, offdps=offdps
    )


def TankMonk(name: str, offhealer=False, offdps=False) -> WoWPlayer:
    return WoWPlayer(
        name=name,
        tankMain=True,
        hasBrez=False,
        hasLust=False,
        offhealer=offhealer,
        offdps=offdps,
    )


def TankDemonHunter(name: str, offdps=False) -> WoWPlayer:
    return WoWPlayer(
        name=name, tankMain=True, hasBrez=False, hasLust=False, offdps=offdps
    )


### Healers
def HealerPaladin(name: str, offtank=False, offdps=False) -> WoWPlayer:
    return WoWPlayer(
        name=name,
        healerMain=True,
        hasBrez=True,
        hasLust=False,
        offtank=offtank,
        offdps=offdps,
    )


def HealerPriest(name: str, offdps=False) -> WoWPlayer:
    return WoWPlayer(
        name=name, healerMain=True, hasBrez=False, hasLust=False, offdps=offdps
    )


def HealerShaman(name: str, offdps=False) -> WoWPlayer:
    return WoWPlayer(
        name=name, healerMain=True, hasBrez=False, hasLust=True, offdps=offdps
    )


def HealerDruid(name: str, offtank=False, offdps=False) -> WoWPlayer:
    return WoWPlayer(
        name=name,
        healerMain=True,
        hasBrez=True,
        hasLust=False,
        offtank=offtank,
        offdps=offdps,
    )


def HealerEvoker(name: str, offdps=False) -> WoWPlayer:
    return WoWPlayer(
        name=name, healerMain=True, hasBrez=False, hasLust=True, offdps=offdps
    )


def HealerMonk(name: str, offtank=False, offdps=False) -> WoWPlayer:
    return WoWPlayer(
        name=name,
        healerMain=True,
        hasBrez=False,
        hasLust=False,
        offtank=offtank,
        offdps=offdps,
    )


### DPS
def DeathKnight(
    name: str,
    offtank=False,
) -> WoWPlayer:
    return WoWPlayer(
        name=name,
        dpsMain=True,
        melee=True,
        hasBrez=True,
        hasLust=False,
        offtank=offtank,
    )


def DemonHunter(
    name: str,
    offtank=False,
) -> WoWPlayer:
    return WoWPlayer(
        name=name,
        dpsMain=True,
        melee=True,
        hasBrez=False,
        hasLust=False,
        offtank=offtank,
    )


def BalanceDruid(name: str, offtank=False, offhealer=False) -> WoWPlayer:
    return WoWPlayer(
        name=name,
        dpsMain=True,
        ranged=True,
        hasBrez=True,
        hasLust=False,
        offhealer=offhealer,
        offtank=offtank,
    )


def FeralDruid(name: str, offtank=False, offhealer=False) -> WoWPlayer:
    return WoWPlayer(
        name=name,
        dpsMain=True,
        melee=True,
        hasBrez=True,
        hasLust=False,
        offhealer=offhealer,
        offtank=offtank,
    )


def Evoker(name: str, offhealer=False) -> WoWPlayer:
    return WoWPlayer(
        name=name,
        dpsMain=True,
        ranged=True,
        hasBrez=False,
        hasLust=True,
        offhealer=offhealer,
    )


def Hunter(
    name: str,
) -> WoWPlayer:
    return WoWPlayer(
        name=name,
        dpsMain=True,
        ranged=True,
        hasBrez=False,
        hasLust=True,
    )


def Mage(
    name: str,
) -> WoWPlayer:
    return WoWPlayer(
        name=name,
        dpsMain=True,
        ranged=True,
        hasBrez=False,
        hasLust=True,
    )


def Monk(name: str, offtank=False, offhealer=False) -> WoWPlayer:
    return WoWPlayer(
        name=name,
        dpsMain=True,
        melee=True,
        hasBrez=False,
        hasLust=False,
        offhealer=offhealer,
        offtank=offtank,
    )


def Paladin(name: str, offtank=False, offhealer=False) -> WoWPlayer:
    return WoWPlayer(
        name=name,
        dpsMain=True,
        melee=True,
        hasBrez=True,
        hasLust=False,
        offhealer=offhealer,
        offtank=offtank,
    )


def Priest(name: str, offhealer=False) -> WoWPlayer:
    return WoWPlayer(
        name=name,
        dpsMain=True,
        ranged=True,
        hasBrez=False,
        hasLust=False,
        offhealer=offhealer,
    )


def Rogue(
    name: str,
) -> WoWPlayer:
    return WoWPlayer(
        name=name,
        dpsMain=True,
        melee=True,
        hasBrez=False,
        hasLust=False,
    )


def Shaman(name: str, offhealer=False) -> WoWPlayer:
    return WoWPlayer(
        name=name,
        dpsMain=True,
        ranged=True,
        hasBrez=False,
        hasLust=True,
        offhealer=offhealer,
    )


def Warlock(name: str) -> WoWPlayer:
    return WoWPlayer(
        name=name,
        dpsMain=True,
        ranged=True,
        hasBrez=False,
        hasLust=False,
    )


def Warrior(name: str, offtank=False) -> WoWPlayer:
    return WoWPlayer(
        name=name,
        dpsMain=True,
        melee=True,
        hasBrez=False,
        hasLust=False,
        offtank=offtank,
    )

File: packages/bot/tests/prebuiltClasses.ts
import { WoWPlayer } from '@mythicplus/shared';

// Tanks
export function TankPaladin(
  name: string,
  { offhealer = false, offdps = false } = {},
): WoWPlayer {
  return WoWPlayer.fromFlags({
    name,
    tankMain: true,
    hasBrez: true,
    offhealer,
    offdps,
  });
}

export function TankWarrior(name: string, { offdps = false } = {}): WoWPlayer {
  return WoWPlayer.fromFlags({ name, tankMain: true, offdps });
}

export function TankDruid(
  name: string,
  { offhealer = false, offdps = false } = {},
): WoWPlayer {
  return WoWPlayer.fromFlags({
    name,
    tankMain: true,
    hasBrez: true,
    offhealer,
    offdps,
  });
}

export function TankDeathKnight(name: string, { offdps = false } = {}): WoWPlayer {
  return WoWPlayer.fromFlags({ name, tankMain: true, hasBrez: true, offdps });
}

export function TankMonk(
  name: string,
  { offhealer = false, offdps = false } = {},
): WoWPlayer {
  return WoWPlayer.fromFlags({ name, tankMain: true, offhealer, offdps });
}

export function TankDemonHunter(name: string, { offdps = false } = {}): WoWPlayer {
  return WoWPlayer.fromFlags({ name, tankMain: true, offdps });
}

// Healers
export function HealerPaladin(
  name: string,
  { offtank = false, offdps = false } = {},
): WoWPlayer {
  return WoWPlayer.fromFlags({
    name,
    healerMain: true,
    hasBrez: true,
    offtank,
    offdps,
  });
}

export function HealerPriest(name: string, { offdps = false } = {}): WoWPlayer {
  return WoWPlayer.fromFlags({ name, healerMain: true, offdps });
}

export function HealerShaman(name: string, { offdps = false } = {}): WoWPlayer {
  return WoWPlayer.fromFlags({ name, healerMain: true, hasLust: true, offdps });
}

export function HealerDruid(
  name: string,
  { offtank = false, offdps = false } = {},
): WoWPlayer {
  return WoWPlayer.fromFlags({
    name,
    healerMain: true,
    hasBrez: true,
    offtank,
    offdps,
  });
}

export function HealerEvoker(name: string, { offdps = false } = {}): WoWPlayer {
  return WoWPlayer.fromFlags({ name, healerMain: true, hasLust: true, offdps });
}

export function HealerMonk(
  name: string,
  { offtank = false, offdps = false } = {},
): WoWPlayer {
  return WoWPlayer.fromFlags({ name, healerMain: true, offtank, offdps });
}

// DPS
export function DeathKnight(name: string, { offtank = false } = {}): WoWPlayer {
  return WoWPlayer.fromFlags({
    name,
    dpsMain: true,
    melee: true,
    hasBrez: true,
    offtank,
  });
}

export function DemonHunter(name: string, { offtank = false } = {}): WoWPlayer {
  return WoWPlayer.fromFlags({ name, dpsMain: true, melee: true, offtank });
}

export function BalanceDruid(
  name: string,
  { offtank = false, offhealer = false } = {},
): WoWPlayer {
  return WoWPlayer.fromFlags({
    name,
    dpsMain: true,
    ranged: true,
    hasBrez: true,
    offhealer,
    offtank,
  });
}

export function FeralDruid(
  name: string,
  { offtank = false, offhealer = false } = {},
): WoWPlayer {
  return WoWPlayer.fromFlags({
    name,
    dpsMain: true,
    melee: true,
    hasBrez: true,
    offhealer,
    offtank,
  });
}

export function Evoker(name: string, { offhealer = false } = {}): WoWPlayer {
  return WoWPlayer.fromFlags({
    name,
    dpsMain: true,
    ranged: true,
    hasLust: true,
    offhealer,
  });
}

export function Hunter(name: string): WoWPlayer {
  return WoWPlayer.fromFlags({
    name,
    dpsMain: true,
    ranged: true,
    hasLust: true,
  });
}

export function Mage(name: string): WoWPlayer {
  return WoWPlayer.fromFlags({
    name,
    dpsMain: true,
    ranged: true,
    hasLust: true,
  });
}

export function Monk(
  name: string,
  { offtank = false, offhealer = false } = {},
): WoWPlayer {
  return WoWPlayer.fromFlags({ name, dpsMain: true, melee: true, offhealer, offtank });
}

export function Paladin(
  name: string,
  { offtank = false, offhealer = false } = {},
): WoWPlayer {
  return WoWPlayer.fromFlags({
    name,
    dpsMain: true,
    melee: true,
    hasBrez: true,
    offhealer,
    offtank,
  });
}

export function Priest(name: string, { offhealer = false } = {}): WoWPlayer {
  return WoWPlayer.fromFlags({ name, dpsMain: true, ranged: true, offhealer });
}

export function Rogue(name: string): WoWPlayer {
  return WoWPlayer.fromFlags({ name, dpsMain: true, melee: true });
}

export function Shaman(name: string, { offhealer = false } = {}): WoWPlayer {
  return WoWPlayer.fromFlags({
    name,
    dpsMain: true,
    ranged: true,
    hasLust: true,
    offhealer,
  });
}

export function Warlock(name: string): WoWPlayer {
  return WoWPlayer.fromFlags({ name, dpsMain: true, ranged: true });
}

export function Warrior(name: string, { offtank = false } = {}): WoWPlayer {
  return WoWPlayer.fromFlags({ name, dpsMain: true, melee: true, offtank });
}
